### ISSUE TITLE
move one beat left does not work

### ISSUE THREAD

**Issue by guiv42** [helge17/tuxguitar#585]
configuration: TuxGuitar 1.6.4, Linux openSuse tumbleweed
bug is reproduced with TuxGuitar 1.5.6

video (1.5.6) is self-explanatory

https://github.com/user-attachments/assets/92a6df63-2e75-4618-a2d3-0b2af314cce9

**Comment by ffio1** [comment]
Proposed fix sent in PR #601.

**Comment by guiv42** [comment]
Before going any further let me ask a trivial but important question: what are "move beat" actions *supposed to do*? It's not that obvious.
Current behavior is measure-wide and seems to be:
- move right: insert one rest (with duration of current note) and shift right all following beats of measure. This can only be done if measure ends with enough rest(s), which are then deleted
- move left: delete current beat, shift all following beats left and insert rest at measure end

Important note: there are not so many controls available in Android ui, but these 2 actions are present and as far as I know nobody complained. Does this mean this behavior is acceptable? No idea.

**Comment by belfie13** [comment]
sounds like move right should be called beat insert rest and move left should be called beat remove note!

**Comment by guiv42** [comment]
> sounds like move right should be called beat insert rest and move left should be called beat remove note!

Yep, it looks like that. But that doesn't make sense as it collides with other items available in the same menu: Beat -> Insert Rest Beat, Beat -> Delete Note or Rest.

I've got to check the implementation

**Comment by guiv42** [comment]
I've written a few erroneous statements above, so let's stick to the code.
Forget Android: I've been confused by the "left/right arrow" icons, these are completely unrelated to "move left/right".

# move left action
- deletes current beat
- moves all following beats *of track* left by the duration of the selected *voice*

Behavior of menu item "Beat -> Delete Note or Rest" is quite different:
- if selected voice contains notes, it deletes the notes
- else, if another voice contains notes, it does nothing
- else if no voice contains note
  - if selected voice is the longest one it does nothing
  - else it removes the beat, moves all following beats *of measure* left, and inserts a rest at measure end

# move right action
I's not so clear. It always inserts a rest beat, with a duration corresponding to selected *voice*. But depending from beat content, this rest may be inserted *before* or *after* the beat (!)

https://github.com/user-attachments/assets/0eaa5a55-f061-45d6-96bb-3f5010bea30e


To sum up: I'm lost.

**Comment by guiv42** [comment]
Maybe I should have red the [doc](https://www.tuxguitar.app/files/1.6.6/desktop/help/detail_measure_beat.html#shift_beats) first. While it seemed a bit counter intuitive to me, I must admit this is pretty well documented.
```
Beat → Move one Beat left: Cleans the beat at the current cursor position in the current track and moves all beats after the cursor one beat to the left.
```

Just forget it.
